### DISCUSSION TITLE
Drift detected in migration after migrating to CockroachDB

### DISCUSSION THREAD

**Discussion by nickhansel** [prisma/prisma#17595] [Q&A] [answered]
### Bug description

1. I am getting "Drift detected: Your database schema is not in sync with your migration history." after running "prisma migrate dev --name init" on a CockroachDB instance with data already in my tables. There are no schema changes as I am attempting to connect to an existing DB.

### How to reproduce

1. prisma init
2.  prisma db pull
3. prisma generate
4. prisma migrate dev --name init

### Expected behavior

I expected nothing to occur because there are no schema changes in my database.

### Environment & setup

- OS: MacOS
- Database: CockroachDB
- Node.js version: 16.17.0

### Prisma Version

```
prisma@4.9.0
```

**Comment by janpio** [comment, marked as answer]
On a new database you ahve to baseline first so the migration history, both the file and in the migration table in the databse, exists: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining

**Reply by nickhansel** [reply]
I am still getting the error even though the databases are completely in sync when trying to baseline.

```
The migrations recorded in the database diverge from the local migrations directory.

✔ We need to reset the CockroachDB database "db" at "db_url"
Do you want to continue? All data will be lost.
```

**Reply by janpio** [reply]
> The migrations recorded in the database diverge from the local migrations directory.

Now it is telling you that the files in `prisma/migration` are different from the ones that have been recorded to be applied. That is an unrecoverable state for the CLI. You can try to manually fix it by modifying the `_prisma_migrations` table.

**Reply by nickhansel** [reply]
Thank you. I resolved this issue by deleting all of the data in the _prisma_migrations table and then baselining the database from there.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
